packages/block-editor/src/store/actions.js

```diff
@@ -1010,3 +1010,35 @@ export function toggleBlockHighlight( clientId, isHighlighted ) {
 		isHighlighted,
 	};
 }
+
+export function __experimentalRegisterTip(
+	scope,
+	context,
+	keywords,
+	description
+) {
+	return {
+		type: 'REGISTER_TIP',
+		scope,
+		context,
+		keywords,
+		description,
+	};
+}
```

Comment:
The more I think about this, the more I lean towards a dedicated package to register and retrieve tips across screens (and not in block-editor directly). A package similar to the keyboard-shortcuts one. 

I'd like @aduth's opinion here too if possible. 

It would also be cool to clarify the meaning of each argument (maybe using typing).
I  assume:

 - scope means the tips "location" (inserter here for example)
 - context I don't know what it is
 - keywords, potential keywords to filter the tips
 - description (or content) is the actual content of the tip

Should we use an object in order to have a more extensible API?

Reply:
> The more I think about this, the more I lean towards a dedicated package to register and retrieve tips across screens (and not in block-editor directly). A package similar to the keyboard-shortcuts one.

Sounds good to me too as long as we agree that the Redux approach could fit well for the Tips feature.
Glad to move it to a dedicated package.

> Should we use an object in order to have a more extensible API?

Probably, yes. Does it mean we could add arbitrary data into the state tree? Or maybe we could an `extra` field in order to store additional data.

> context I don't know what it is

It's a good example of additional data required only for some tips. In this case, I used it as a way to define tip context in the block inserter menu. Its values could be `CSS`, `header`, `theme`, `plugin` and `color`. Not totally needed to be honest. We can get rid of it.

Reply:
Could the [`@wordpress/notices` package](https://github.com/WordPress/gutenberg/tree/master/packages/notices) serve as prior art for this sort of requirement? Notices are somewhat similar in being data which represents a text-like representation _somewhere_ in the page. Notices were even modeled to support multiple "groupings" / "areas" where they could be assigned (`context`), though we don't currently leverage it as far as I'm aware.